### PR TITLE
Improve toList and ModelValue type usage

### PR DESCRIPTION
**Pull request by lindner**
- Returns ModelValue[] from toList
- Adjusts rawData to be a property object, adjusts tests to use it.
- Make SyntheticStorage compliant and fix more type issues
- Fix bad type usage in DescrptionFormatter

### CODE REVIEW

**Comment by lindner** [comment]
Note that this is extracted from my changes for #2863  More changes with this as base PR coming...

File: src/runtime/test/data-layer-test.ts [hunk 1]
```diff
@@ -37,6 +37,8 @@ describe('entity', async () => {
     const clone = list[0];
     assert.isDefined(clone);
     assert.deepEqual(clone.rawData, {value: 'hello world'});
-    assert.notEqual(entity, clone);
+
```
Comment (lindner) [inline src/runtime/test/data-layer-test.ts]: This does not appear to be a valid test since the two objects are not comparable...

Comment (shans) [reply src/runtime/test/data-layer-test.ts]: Once we provide the full storage stack to the shell then we should be able to use a similar stack here, at which point the types will be the same. Not sure of the value overall of checking they're not deeply equal though - probably just looking at the ids would be sufficient.

Comment (lindner) [reply src/runtime/test/data-layer-test.ts]: The IDs are the same.  I'll add a TODO to this for future investigation.

**Pull request merged**